Make OP_DEFINED eligible for constant folding

richardleach opened the pull request:
Prior to this, the condition in the following example did not fold:

    'use constant ONE => 1; my $x = (defined(ONE)) ? 1 : 0;'

With this commit, the final op tree produced for the above is just:

    5  <@> leave[1 ref] vKP/REFC ->(end)
    1     <0> enter v ->2
    2     <;> nextstate(main 193 -e:1) v:%,us,{,fea=15 ->3
    4     <1> padsv_store[$x:193,194] vKS/LVINTRO ->5
    3        <$> const[IV 1] s/FOLD ->4
    -        <0> ex-padsv sRM*/LVINTRO ->4

Closes #16266 

---------------------------------------------------------------------------------
* This seems like a bit of an edge case, so I don't think it needs a perldelta entry.

richardleach committed:
Make OP_DEFINED eligible for constant folding

Prior to this, the condition in the following example did not fold:

    'use constant ONE => 1; my $x = (defined(ONE)) ? 1 : 0'

With this commit, the final op tree produced for the above is just:

    5  <@> leave[1 ref] vKP/REFC ->(end)
    1     <0> enter v ->2
    2     <;> nextstate(main 193 -e:1) v:%,us,{,fea=15 ->3
    4     <1> padsv_store[$x:193,194] vKS/LVINTRO ->5
    3        <$> const[IV 1] s/FOLD ->4
    -        <0> ex-padsv sRM*/LVINTRO ->4

tonycoz commented:
The op tree in the commit message doesn't match the perl code (where did `$x` come from?)

richardleach commented:
> The op tree in the commit message doesn't match the perl code (where did `$x` come from?)

Sorry about that. I changed examples a couple of times while writing. Now updated to match.

richardleach commented:
@tonycoz - am I okay to merge this?
